core/runtime/common/runtime_context.cpp

```diff
@@ -50,12 +50,13 @@ namespace kagome::runtime {
 
   outcome::result<RuntimeContext> RuntimeContextFactory::fromCode(
       const runtime::ModuleFactory &module_factory,
+      const WasmInstrumenter &instrumenter,
       common::BufferView code_zstd,
       ContextParams params) {
```

Comment:
These params should hide inside `RuntimeInstancesPool`. See #2140
```suggestion
      std::shared_ptr<ModuleInstance> instance) {
```

Reply:
The goal here is to create an instance without pooling it, so whatever's hidden inside RuntimeInstancesPool stays here, and I pass the dependencies for instantiation separately.